src/main/java/uk/gov/hmcts/reform/sscs/ccd/service/UpdateCcdCaseService.java

```diff
@@ -31,6 +34,30 @@ public UpdateCcdCaseService(IdamService idamService,
         this.ccdClient = ccdClient;
     }
 
+    @Retryable
+    public SscsCaseDetails updateCaseV2(Long caseId, String eventType, String summary, String description, IdamTokens idamTokens, Consumer<SscsCaseData> mutator) {
+        log.info("UpdateCaseV2 for caseId {} and eventType {}", caseId, eventType);
+        StartEventResponse startEventResponse = ccdClient.startEvent(idamTokens, caseId, eventType);
+        var data = sscsCcdConvertService.getCaseData(startEventResponse.getCaseDetails().getData());
+        mutator.accept(data);
+        CaseDataContent caseDataContent = sscsCcdConvertService.getCaseDataContent(data, startEventResponse, summary, description);
+
+        return sscsCcdConvertService.getCaseDetails(ccdClient.submitEventForCaseworker(idamTokens, caseId, caseDataContent));
+    }
+
+    @Retryable
+    public SscsCaseDetails updateCaseV2(Long caseId, String eventType, IdamTokens idamTokens, Function<SscsCaseData, String[]> mutator) {
```

Comment:
Rather than using a String[] I suggest returning a type here - something like UpdateResponse with description and summary.

Suggest using Java's new record types - 

`    record UpdateResult(String summary, String description) { }`

Then you can use it like

```
var result = mutator.apply(data);
CaseDataContent caseDataContent = sscsCcdConvertService.getCaseDataContent(data, startEventResponse, result.summary, result.description);

```